src/main/java/org/jabref/gui/entryeditor/SourceTab.java

```diff
@@ -77,6 +78,15 @@ private Node createSourceEditor(BibEntry entry, BibDatabaseMode mode) {
             LOGGER.debug("Incorrect entry", ex);
         }
 
+        codeArea.richChanges().subscribe(x -> {
```

Comment:
@tobiasdiez  I now found a way to get position information when some data is changed in the code editor, e.g. exactly the information I would need for the sharejs thing (I need the range information (e.g. at position 3 the character x was deleted)

Two problems:
1. This only applies to one entry - whereas I would need a similar function for the whole database content
The problem is that a simple value changed event is not sufficient, because there I only get the old and new value.
However, there is something like https://docs.oracle.com/javase/8/javafx/api/javafx/scene/control/TextFormatter.Change.html 
which applies for any controls that have TextInput


2.  This even only fires when I change sth in the code editor and I am not sure if it would be fired if I edit just the author.

Maybe you have some idea?